### ISSUE TITLE
False positive for `Loop which does not loop`

### ISSUE THREAD

**Issue by zerkms** [kalessil/phpinspectionsea#1634]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) v4.0.6 |
| Language level | PHP 7.2                      |

Current behaviour
--
```php
<?php

function f(): bool {
    // side effects
    return true;
}

while (f()) {};
```

In this snipped `while` keyword is marked as "This loop does not loop", while it certainly does.

Logic behind this code is that `f()` must run until it returns `false` and while it makes some useful side effects.

Expected behaviour
--
It should not trigger the warning.

Environment details
---

```
PhpStorm 2020.2.1
Build #PS-202.6948.87, built on August 27, 2020
Licensed to <stripped>
You have a perpetual fallback license for this version
Subscription is active until <stripped>
Runtime version: 11.0.8+10-b944.31 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 4.15.0-117-generic
GC: ParNew, ConcurrentMarkSweep
Memory: 1700M
Cores: 6
Registry: run.processes.with.pty=TRUE
Non-Bundled Plugins: YAML/Ansible support, name.kropp.intellij.makefile, net.seesharpsoft.intellij.plugins.csv, mobi.hsz.idea.gitignore, com.intellij.kubernetes, com.dmarcotte.handlebars, com.kalessil.phpStorm.phpInspectionsEA, de.espend.idea.php.toolbox, de.espend.idea.php.annotation, de.espend.idea.php.generics, fr.adrienbrault.idea.symfony2plugin, ru.adelf.idea.dotenv
Current Desktop: ubuntu:GNOME
```
